src/megaapi_impl.cpp

```diff
@@ -378,6 +378,20 @@ MegaNode *MegaNodePrivate::copy()
     return new MegaNodePrivate(this);
 }
 
+char *MegaNodePrivate::serialize()
+{
+    string d;
+    if (!serialize(&d))
+    {
+        return NULL;
+    }
+
+    char *ret = new char[d.size()*4/3+3];
```

Comment:
I believe it to be safer with `+ 4`:
`char *ret = new char[d.size() * 4 / 3 + 4];`